wp-lazy-loading.php

```diff
@@ -94,38 +94,32 @@ function _wp_lazy_loading_add_attribute_to_attachment_image( $attr ) {
 // The following functions will be merged to core.
 
 /**
- * Get the HTML tags to lazy-load.
+ * Determine whether to add the `loading` attribute to the specified tag in the specified context.
  *
  * @since (TBD)
  *
- * @return array List of tags to add loading="lazy" attributes to.
+ * @param string  $tag_name The tag name.
+ * @param string  $context Additional context, like the current filter name or the function name from where this was called.
+ * @return boolean Whether to add the attribute.
  */
-function wp_get_lazy_load_tags() {
-	// See https://github.com/whatwg/html/issues/2806.
-	$supported_tags = array( 'img', 'iframe' );
+function wp_add_lazy_loading_to( $tag_name, $context ) {
+	// By default add to all 'img' tags.
+	// See https://github.com/whatwg/html/issues/2806
+	$add = ( 'img' === $tag_name );
 
 	/**
-	 * Filters on which HTML tags to add `loading="lazy"`.
+	 * Filters whether to add the `loading` attribute to the specified tag in the specified context.
 	 *
 	 * @since (TBD)
 	 *
-	 * @param array $tags List of tags to add loading="lazy" attributes to. Default is an array with 'img'.
-	 *                    Supports passing a boolean: `false` removes all tags, `true` resets the tags to the default value.
-	 *                    Example: `add_filter( 'wp_get_lazy_load_tags', '__return_true' );` will enable it for images and iframes.
-	 *                    Note that the HTML specs are not finalized yet. See https://github.com/whatwg/html/pull/3752/files,
-	 *                    and support for iframes has been postponed (for now).
+	 * @param boolean $add Defatls value.
+	 * @param string  $tag_name The tag name.
+	 * @param string  $context Additional context, like the current filter name or the function name from where this was called.
 	 */
-	$tags = apply_filters( 'wp_get_lazy_load_tags', array( 'img' ) );
-
-	// Support a boolean, false to disable, true to reset to default.
-	if ( ! is_array( $tags ) ) {
-		return $tags === true ? $supported_tags : array();
-	}
-
-	// Only include supported tags.
-	return array_values( array_intersect( $tags, $supported_tags ) );
+	return (bool) apply_filters( 'wp_add_lazy_loading_to', $add, $tag_name, $context );
```

Comment:
The same applies to the filter of course, it should have the same name we choose for the function (which it does currently, but I think we should change that name).

Reply:
Yep, of course.